### ISSUE TITLE
[Feature request] Add Support of Encrypting/Decryptiong GPG/PGP Fields to Aviod Single Point of Failure

### ISSUE THREAD

**Issue by gynet** [keeweb/keeweb#1538]
First, thanks for the very nice job did for the YubiKey integration, the CR mode would help a lot for some thread modeling.

However, as we know the CR mode is not a truly 2FA because of the nature of an offline database application, that means if one specific version of the database file + the password  + CR response + (key file) got compromised at the same time, then all the entries in the database could be exposed to an attacker, also as most of the cloud service has a version tracking system, all the versions of the database are already recoverable.

 I understand this is not only a unique problem to the KeePass ecosystem but also almost all the password manager applications, but it’s somehow a “Single Point of Failure” situation.

On the contrary, the [Passwordstore](https://www.passwordstore.org/) app uses individual GPG encrypted files to save all the passwords, also, as you might know, Yubikey can to be used as a GPG smart card for encryption/decryption, so that even the whose user system is compromised, the attacker can only acquire only the password which is decrypted, and other passwords entries remain encrypted. The decryption process only happens on the Yubikey side, and the private key is not exposed to the system at all. I believe it will resolve the SPOF problem.  However, the implementation of Passwordstore does not encrypted metadata, which I think is a vulnerability, although they can add a tomb as an encrypted wrapper on top, it’s lack of cross-platform support. 

Compare to Passwordstore, I still prefer KeeWeb/KeePass because of the UX and encrypted metadata. My current workaround of SPOF is to manually encrypt some fields using OpenGPG, and I wrote a BASH script run with auto-type to decrypt the field. It's not bad, but it's kind of awkward. 

**Describe the solution you'd like**
So, if KeeWeb can add support(maybe a plugin?) of encrypting /decrypting fields using GPG, it would be super awesome, as it fully resolves both of problem of SPOF and the metadata, and the GPG encrypted fields can be played as secondary encryption layer, and metadata has already fully encrypted by KeePass's algorithm, what's more, the users can use the YubiKey a truly “2FA”.

**Comment by gynet** [comment]
It would nice to see something like below can be supported by KeeWeb using a right-click menu to decrypt as well as the creation 
![image](https://user-images.githubusercontent.com/2174522/84557413-ce79e180-acdf-11ea-8adb-3803b07cb2a5.png)

**Comment by gynet** [comment]
Some user scenarios: 
- Someone posted a [similar idea](https://github.com/keepassxreboot/keepassxc/issues/4064) for a feature request to KeepassXC, but the user asks was if it is possible to encrypt entire entry which is not possible because of compatibility. However, encrypting a field should be possible and compatible. 
- I also believe this [user](https://github.com/keeweb/keeweb/issues/1307) might use a workaround similar to mine.

**Comment by antelle** [comment]
Hi! There are two things:
- it's not possible to touch a yubikey for GPG encryption/decryption, or if it's possible probably nobody does this because it's used in other places as well, but maybe that's tolerable depending on a thread model; 
- we need support from other clients to do this, otherwise the file won't be compatible, well, it will be compatible, but it won't give any password. Is there a similar issue in KeePassXC and any of major iOS clients (Strongbox or KeePassium)?

**Comment by gynet** [comment]
@antelle 
- You can enable touch on by using [ykman](https://support.yubico.com/support/solutions/articles/15000012643-yubikey-manager-cli-ykman-user-manual#ykman_openpgp_set-touchjj4mga) 
I just tried the below command, it works! 
   ```
     ykman openpgp set-touch [OPTIONS] KEY POLICY
   ```
Even without touch, the user still need to input the Yubikey physically and type the pin 
- Agree, but the good thing is GPG is an open-source standard, there are already pretty good mobile GPG app, e.g for android, I use OpenKeychain, it can decrypt a file using Yubikey's NFC, and encryption and decryption communication can go through Androids ["Interacting with Other Apps" APIs](https://developer.android.com/training/basics/intents) or just the clipboard. 

For native app integration, it's a might be a chicken or egg situation, although it would be ideal to also have an extension to a mobile app, e.g, Keepass2Adnroid, so far, I would say combine OpenKeychain is pretty easy to use.

**Comment by antelle** [comment]
I also agree that it's a chicken and egg problem and probably if there's support in KeeWeb and an extension for KeePass which is not hard to create, it may get some wider support in other forks.
I wonder if there will be support on iOS, I haven't seen any app capable to use GPG with a yubikey, however [this issue](https://github.com/Yubico/yubikit-ios/issues/19) suggests it's not impossible, albeit not very straightforward.
Overall, I gave it a round of thought and it seems to be quite an interesting mode that can be used as an alternative to the current implementation for those who need it. Here are some considerations, but the list is indeed not complete and not very well made, anyway here it is:
- this can be enabled either for the whole file, or individual for entries, or for groups with inheritance, we need to weigh up pros and cons and decide which mode would be flexible enough; 
- there's search by protected fields in different forks including KeeWeb; it won't work and this most likely needs to be explained in the search control if the feature is in use; 
- db export may become slower which is fine, however if the touch feature is enabled, it won't be possible (or inhumanly painful); 
- the same applies to disabling the feature.
